Source/WebCore/html/HTMLSelectElement.cpp

```diff
@@ -432,18 +432,6 @@ void HTMLSelectElement::optionElementChildrenChanged()
         cache->childrenChanged(this);
 }
 
-void HTMLSelectElement::setMultiple(bool multiple)
-{
-    bool oldMultiple = this->multiple();
-    int oldSelectedIndex = selectedIndex();
-    setBooleanAttribute(multipleAttr, multiple);
-
-    // Restore selectedIndex after changing the multiple flag to preserve
-    // selection as single-line and multi-line has different defaults.
-    if (oldMultiple != this->multiple())
-        setSelectedIndex(oldSelectedIndex);
```

Comment:
Since I don't think it's safe to remove the WebKitLegacy method, we should keep this method around.

Though it can just call `setBooleanAttribute(multipleAttr, multiple);` now.

Reply:
Can't we just call `IMPL->setBooleanAttribute(multipleAttr, newMultiple);` directly from WebKitLegacy instead of keeping this around?

Reply:
Good point! Let's do that.

Reply:
@nt1m - Thanks for guidance. Pushed changes in new commit.